codec/src/main/java/io/netty/handler/codec/MessageToMessageDecoder.java

```diff
@@ -66,9 +69,49 @@ protected MessageToMessageDecoder() {
      * @param inboundMessageType    The type of messages to match and so decode
      */
     protected MessageToMessageDecoder(Class<? extends I> inboundMessageType) {
+        this(true, inboundMessageType);
+    }
+
+    /**
+     * Create a new instance which will try to detect the types to match out of the type parameter of the class.
+     *
+     * @param allowSharable         if this decoder is not sharable it will take care of calling
+     *                              {@link ChannelHandlerContext#read()} if non message was produced and
+     *                              {@link io.netty.channel.ChannelOption#AUTO_READ} is set to {@code false}.
+     */
+    protected MessageToMessageDecoder(boolean allowSharable) {
+        if (!allowSharable) {
+            ensureNotSharable();
```

Comment:
`ensureNotSharable()` is calling `isSharable()` before `this.allowSharable` is set, so it's always going to succeed and therefore is not very useful.

Reply:
good catch